### PR TITLE
add spinner in saving project, can avoid multiple commit 

### PR DESCRIPTION
**Pull request by yongbing-chen**

### CODE REVIEW

File: src/react/components/pages/projectSettings/projectSettingsPage.tsx [hunk 1]
```diff
@@ -176,29 +186,34 @@ export default class ProjectSettingsPage extends React.Component<IProjectSetting
...
-        } else {
-            this.props.history.goBack();
+            if (isNew) {
+                this.props.history.push(`/projects/${this.props.project.id}/edit`);
+            } else {
+                this.props.history.goBack();
+            }
+        } finally {
+            this.setState({ isCommiting: false });
         }
```
Comment (stew-ro) [inline src/react/components/pages/projectSettings/projectSettingsPage.tsx:215]: Why do we need a try here?

Comment (yongbing-chen) [reply src/react/components/pages/projectSettings/projectSettingsPage.tsx:215]: there is a try... finally, ensure **_this.setState({ isCommiting: false });_** will execute.

**Review by stew-ro — COMMENTED** [review]
Now if user tries to create project with name that already existst, all input fields are reset
image

**Review by stew-ro — CHANGES_REQUESTED** [review]
Persist user input when project already exists

**Comment by yongbing-chen** [comment]
> Persist user input when project already exists

fixed, please help test

**Comment by stew-ro** [comment]
> > Persist user input when project already exists
> 
> fixed, please help test again, thanks

This now reset's all the input fields that user inputted/changed when invalid:
ezgif com-video-to-gif

 Can we keep existing behavior and keep user inputted fields?

**Comment by yongbing-chen** [comment]
> > > Persist user input when project already exists
> > 
> > 
> > fixed, please help test again, thanks
> 
> This now reset's all the input fields that user inputted/changed when invalid:
> ezgif com-video-to-gif
> 
> Can we keep existing behavior and keep user inputted fields?

**Review by stew-ro — APPROVED** [review]

**Pull request merged**